src/Components/Search/Spike/SearchBarSpike.tsx

```diff
@@ -0,0 +1,205 @@
+import { SearchBarSpike_viewer } from "__generated__/SearchBarSpike_viewer.graphql"
+import { SearchBarSpikeSuggestQuery } from "__generated__/SearchBarSpikeSuggestQuery.graphql"
+import { ContextConsumer, ContextProps } from "Artsy"
+import { SearchPreview } from "Components/Search/Previews"
+import { throttle } from "lodash"
+import React, { Component } from "react"
+import Autosuggest from "react-autosuggest"
+import {
+  createRefetchContainer,
+  graphql,
+  QueryRenderer,
+  RelayRefetchProp,
+} from "react-relay"
+import { get } from "Utils/get"
+
+export interface Props extends ContextProps {
+  relay: RelayRefetchProp
+  viewer: SearchBarSpike_viewer
+}
+
+interface State {
+  input: string
+  entityID: string
+  entityType: string
+}
+
+export class SearchBarSpike extends Component<Props, State> {
+  state = {
+    input: "",
+    entityID: null,
+    entityType: null,
+  }
+
+  throttledFetch = term => {
+    this.props.relay.refetch(
+      {
+        term,
+        hasTerm: true,
+      },
+      null,
+      error => {
+        if (error) console.error(error)
+      }
+    )
+  }
+
+  componentDidMount() {
+    this.throttledFetch = throttle(this.throttledFetch, 500, {
+      leading: true,
+    })
+  }
+
+  searchTextChanged = (_e, { newValue: input }) => {
+    this.setState({ input })
+  }
+
+  renderPreview() {
+    const { entityID, entityType } = this.state
+    if (entityID && entityType) {
+      return <SearchPreview entityID={entityID} entityType={entityType} />
+    }
+  }
+
+  onSuggestionHighlighted = ({ suggestion }) => {
+    if (!suggestion) return null
+    const {
+      node: { searchableType: entityType, id: entityID },
+    } = suggestion
+    this.setState({ entityType, entityID })
+  }
+
+  renderSuggestionsContainer = ({ containerProps, children, query }) => {
+    if (!query) return null
+    return (
+      <Box {...containerProps}>
+        <Box>Search "{query}"</Box>
+        {children}
+      </Box>
+    )
+  }
+
+  render() {
+    const { viewer } = this.props
+    const edges = get(viewer, v => v.search.edges, [])
+
+    const inputProps = {
+      onChange: this.searchTextChanged,
+      placeholder: "Search...",
+      value: this.state.input,
+    }
+
+    return (
+      <Flex flexDirection={["column", "row"]}>
+        <Box width={["100%", "25%"]}>
+          <Flex flexDirection="column">
+            <Autosuggest
+              suggestions={edges}
+              onSuggestionsClearRequested={() => {
+                this.setState({ input: "", entityID: null, entityType: null })
+              }}
+              onSuggestionHighlighted={this.onSuggestionHighlighted}
+              onSuggestionsFetchRequested={({ value }) =>
+                this.throttledFetch(value)
+              }
+              getSuggestionValue={({ node: { displayLabel } }) => displayLabel}
+              renderSuggestion={({ node }, { isHighlighted }) => {
+                return (
+                  <span
```

Comment:
This is of course going to change in an upcoming PR with a proper `SuggestionItem` component. This was quick to get the highlighting working.